### ISSUE TITLE
Bindgen layout tests fail on i686 with leptonica 1.84

### ISSUE THREAD

**Issue by NoisyCoil** [ccouzens/leptonica-sys#29]
Hi. Tests for leptonica-sys fail with leptonica 1.84 on i686 due to a known bug reported to bindgen, see https://github.com/rust-lang/rust-bindgen/issues/1823. Apparently, starting from v1.84, the leptonica headers contain a definition of `max_align_t` which has incompatible definitions in C and clang on i686, and thus makes the relative test fail with the error

```
---- bindgen_test_layout_max_align_t stdout ----
thread 'bindgen_test_layout_max_align_t' panicked at 'assertion failed: `(left == right)`
  left: `16`,
 right: `24`: Size of: max_align_t'
```

For the tests I used libleptonica-dev from Debian testing. The bindgen maintainers suggest either making `max_align_t` opaque or blacklisting it from the bindings, depending on whether `max_align_t` is actually used or not in the bindings.

I understand the last version of leptonica supported by this library should be 1.83. However, especially since starting from bindgen v0.70 layout tests became compile-time tests (so this bug would make compiling the library fail), I thought you may want to have a look at this nonetheless.

**Comment by plugwash** [comment]
A possible fix/workaround is to blocklist the type, it doesn't appear to be actually used by leptonica, just picked up from a system header.
[1002_blocklist_max_align_t.patch.txt](https://github.com/user-attachments/files/17912212/1002_blocklist_max_align_t.patch.txt)

**Comment by ccouzens** [comment]
Thank you both

I've accepted the patch to blocklist the type. It is also what I would have done. I've published as 0.4.9.
